script/deploy/healthCheck.ts

```diff
@@ -239,278 +239,356 @@ const main = defineCommand({
     }
 
     //          ╭─────────────────────────────────────────────────────────╮
-    //          │                   Check approved DEXs                   │
+    //          │                   Check whitelisted addresses           │
     //          ╰─────────────────────────────────────────────────────────╯
-    if (dexs) {
-      consola.box('Checking DEXs approved in diamond...')
+    if (expectedWhitelistedAddresses) {
+      consola.box('Checking whitelisted addresses in diamond...')
 
-      // connect with diamond to get whitelisted DEXs
-      const dexManager = getContract({
+      // connect with diamond to get whitelisted addresses
+      const whitelistManager = getContract({
         address: deployedContracts['LiFiDiamond'],
         abi: parseAbi([
-          'function approvedDexs() external view returns (address[])',
-          'function isFunctionApproved(bytes4) external returns (bool)',
+          'function getWhitelistedAddresses() external view returns (address[])',
+          'function isFunctionApproved(bytes4) external view returns (bool)',
+          'function getApprovedFunctionSignatures() external view returns (bytes4[])',
         ]),
         client: publicClient,
       })
 
-      const approvedDexs = await dexManager.read.approvedDexs()
-
-      let numMissing = 0
+      let onChainWhitelisted: Address[] = []
+      try {
+        onChainWhitelisted =
+          await whitelistManager.read.getWhitelistedAddresses()
+      } catch (error) {
+        logError('Failed to get whitelisted addresses from chain')
+        // Don't skip the checks - we'll still check against the config
+      }
 
-      // Check for each address in dexs.json if it is whitelisted
-      for (const dex of dexs) {
-        if (!dex) {
-          logError(`Encountered undefined DEX address.`)
-          continue
-        }
+      // Check if there are any expected whitelisted addresses in config
+      if (
+        expectedWhitelistedAddresses &&
+        expectedWhitelistedAddresses.length > 0
+      ) {
+        let numMissing = 0
+
+        // If we couldn't get on-chain addresses, we can only check the config addresses
+        if (onChainWhitelisted.length === 0) {
+          logError('Cannot verify whitelist status - on-chain data unavailable')
+          for (const cfgAddress of expectedWhitelistedAddresses) {
+            if (!cfgAddress) {
+              logError(`Encountered undefined whitelisted address in config`)
+              continue
+            }
+            try {
+              const normalized = getAddress(cfgAddress)
```

Comment:
to make this more meaningful we could check here if that address is a contract, since we only want to whitelist contract addresses (same in else branch)

Reply:
good point, updated.

The limitation are EOAs with code.

feel free to resolve

Reply:
dont think we gonna whitelist an EOA with code